wagtail/models/__init__.py

```diff
@@ -2976,21 +3025,8 @@ def user_has_lock(self):
         return self.page.locked_by_id == self.user.pk
 
     def page_locked(self):
-        current_workflow_task = self.page.current_workflow_task
-        if current_workflow_task:
-            if current_workflow_task.page_locked_for_user(self.page, self.user):
-                return True
-
-        if not self.page.locked:
-            # Page is not locked
-            return False
-
-        if getattr(settings, "WAGTAILADMIN_GLOBAL_PAGE_EDIT_LOCK", False):
-            # All locks are global
-            return True
-        else:
-            # Locked only if the current user was not the one who locked the page
-            return not self.user_has_lock()
+        lock = self.page.get_lock()
+        return lock and lock.for_user(self.user)
```

Comment:
This should always return a boolean, I think?

```suggestion
        return lock is not None and lock.for_user(self.user)
```